### PR TITLE
Add server address CLI argument

### PR DESCRIPTION
**Pull request by lolski**
## Release notes: product changes
Make server address configurable through command line argument.

## Motivation
User will want to configure the server listen address.

## Implementation
- Add server address command line argument

### CODE REVIEW

File: server/parameters/config.rs [hunk 1]
```diff
@@ -41,23 +41,25 @@ impl Config {
...
     }
 
     pub fn new_with_diagnostics_config(diagnostics_config: DiagnosticsConfig, is_development_mode: bool) -> Self {
-        Self::customised(None, Some(diagnostics_config), None, is_development_mode)
+        Self::customised(None, None, Some(diagnostics_config), None, is_development_mode)
     }
 
     pub fn new_with_data_directory(data_directory: &Path, is_development_mode: bool) -> Self {
-        Self::customised(None, None, Some(data_directory.to_path_buf()), is_development_mode)
+        Self::customised(None, None, None, Some(data_directory.to_path_buf()), is_development_mode)
```
Comment (farost) [inline server/parameters/config.rs:52]: We do need to refactor it at some point. Somehow... It's getting out of control. Now a priority now.

File: server/parameters/cli.rs [hunk 2]
```diff
@@ -3,14 +3,19 @@
...
+use std::net::SocketAddr;
 use clap::{ArgAction, Parser};
 use resource::constants::server::MONITORING_DEFAULT_PORT;
 
 /// TypeDB Core usage
 #[derive(Parser, Debug)]
 #[command(version, about, long_about = None)]
 pub struct CLIArgs {
+
+    /// Server host and port, eg., 0.0.0.0:1729
```
Comment (farost) [inline server/parameters/cli.rs:15]: Let's mention that localhost is temporarily not working

**Review by farost — APPROVED** [review]

File: server/typedb.rs [hunk 3]
```diff
@@ -227,6 +230,14 @@ fn synchronize_database_metrics(diagnostics_manager: Arc<DiagnosticsManager>, da
     diagnostics_manager.submit_database_metrics(metrics);
 }
 
+async fn resolve_address(address: String) -> SocketAddr {
+    lookup_host(address.clone())
+        .await
+        .unwrap()
+        .next()
+        .expect(format!("Unable to map address '{}' to any IP addresses", address).as_str())
```
Comment (farost) [inline server/typedb.rs:238]: It should probably return an Error (`Result<SocketAddr>`) instead, but we can continue with this, but at least a todo.

File: main.rs [hunk 4]
```diff
@@ -39,7 +40,7 @@ async fn main() {
     }
 }
 
-fn get_configuration(cli_args: CLIArgs) -> Config {
+async fn get_configuration(cli_args: CLIArgs) -> Config {
```
Comment (farost) [inline main.rs:43]: Is it an outdated `async`?

File: main.rs [hunk 5]
```diff
@@ -28,7 +29,7 @@ async fn main() {
     initialise_logging_global();
 
     let deployment_id = None;
-    let config = get_configuration(cli_args);
+    let config = get_configuration(cli_args).await;
```
Comment (farost) [inline main.rs:32]: Is it an outdated `await`?

File: server/parameters/config.rs (hunk 1 above)
Comment (lolski) [reply server/parameters/config.rs:52]: Yeah I've got an idea as to how it should look like

File: server/typedb.rs (hunk 3 above)
Comment (lolski) [reply server/typedb.rs:238]: Given that this is not a generic method, but a specific one located in the server, I believe the method has enough context to decide to resolve it there and then.

File: main.rs (hunk 4 above)
Comment (lolski) [reply main.rs:43]: You

File: main.rs (hunk 5 above)
Comment (lolski) [reply main.rs:32]: You're correct again.

**Pull request merged**